bcipy/helpers/task.py

```diff
@@ -370,186 +373,159 @@ def get_key_press(
+                     targets_included: Set[str] = set(["target", "nontarget"]),
+                     targets_excluded: Set[str] = set([])) -> Tuple[np.ndarray, np.ndarray]:
+    """Extract inquiry data and labels.
+
+    Args:
+        trial_target_info (List[str]): labels each trial as "target", "non-target", "first_pres_target", etc
+        timing_info (List[float]): Timestamp of each event in seconds
+        eeg_data (np.ndarray): shape (channels, samples) preprocessed EEG data
+        fs (int): sample rate of EEG data. If data is downsampled, the sample rate should be also be downsampled.
+        trials_per_inquiry (int): number of trials in each inquiry
+        offset (float, optional): Any calculated or hypothesized offsets in timings. Defaults to 0.
+        channel_map (List[int], optional): Describes which channels to include or discard. Defaults to DEFAULT_CHANNEL_MAP.
+        trial_length (float, optional): [description]. Defaults to 0.5.
+        targets_included (Set[str]): target labels to include
+        targets_excluded (Set[str]): target labels to exclude
+
+    Returns:
+        reshaped_data (np.ndarray): inquiry data of shape (Channels, Inquiries, Samples)
+        labels (np.ndarray): integer label for each inquiry. With trials_per_inquiry=K,
+            a label of [0, K-1] indicates position of "target", or label of K indicates
+            target was not present.
+    """
+    # Remove the channels that we are not interested in
+    channels_to_remove = [idx for idx, value in enumerate(channel_map) if value == 0]
+    eeg_data = np.delete(eeg_data, channels_to_remove, axis=0)
+
+    # Number of samples we are interested per inquiry
+    num_samples = int(trial_length * fs * trials_per_inquiry)
+
+    # Remove unwanted elements from target info and timing info
+    tmp_targets, tmp_timing = [], []
+    for target, timing in zip(trial_target_info, timing_info):
+        if target in targets_included and target not in targets_excluded:
+            tmp_targets.append(target)
+            tmp_timing.append(timing)
+    trial_target_info = tmp_targets
+    timing_info = tmp_timing
+
+    n_inquiry = len(timing_info) // trials_per_inquiry
+
+    # triggers in seconds are mapped to triggers in number of samples.
+    triggers = list(map(lambda x: int((x + offset) * fs), timing_info))
+
+    # shape (Channels, Inquiries, Samples)
+    reshaped_data = np.zeros((len(eeg_data), n_inquiry, num_samples))
+
+    # Label for every inquiry
+    labels = np.zeros(n_inquiry)
+
+    for inquiry_idx, chunk in enumerate(grouper(zip(trial_target_info, triggers), trials_per_inquiry)):
+        # label is the index of the "target", or else the length of the inquiry
+        label = trials_per_inquiry
+        first_trigger = None
+        for trial_idx, (target, trigger) in enumerate(chunk):
+            if first_trigger is None:
+                first_trigger = trigger
+
+            if target == 'target':
```

Comment:
What I had in mind was defining labels to include (target, nontarget) & exclude (fixation), and then another place to define a target label as you hardcode here (optional, since this won't be available when we reshape a free spell task for instance). Then you could reshape various labels and assign targetness without hardcoding your assumption about what that label may be.